src/explore-education-statistics-frontend/src/modules/find-statistics/FindStatisticsPage.tsx

```diff
@@ -54,35 +37,34 @@ export const getServerSideProps: GetServerSideProps<Props> = async ({
 }) => {
   const { newDesign } = query;
 
-  // TODO EES-3517 - remove newDesign check
-  const publicationsResponse = newDesign
-    ? await publicationService.listPublications(
-        createPublicationListRequest(query),
-      )
-    : {
-        results: [],
-        paging: {
-          page: 0,
-          pageSize: 0,
-          totalResults: 0,
-          totalPages: 0,
-        },
-      };
-
-  // TODO EES-3517 - remove newDesign check
+  // TODO EES-3517 - remove this
   const themes = newDesign
-    ? await themeService.listThemes()
+    ? []
     : await publicationService.getPublicationTree({
         publicationFilter: 'FindStatistics',
       });
 
+  const queryClient = new QueryClient();
+
+  // TODO EES-3517 - remove newDesign check
+  if (newDesign) {
+    await Promise.all([
+      queryClient.prefetchQuery(['listPublications', query], () =>
+        publicationService.listPublications(
+          createPublicationListRequest(query),
+        ),
+      ),
```

Comment:
Once we start using React Query more, managing query and their keys can be a bit problematic as it can lead to incorrect fetching from the cache if there are naming collisions.

We also want to coordinate keys/queries so that:

- we can re-use them easily in multiple places (e.g. in the server and client)
- they can be easily refactored in the future

There's a few ways to approach this, but the recommendation from the docs is to use the query key factory pattern. 

At it's crux, it consists of something like this:

```ts
// queries/publications.ts
import { useQuery, UseQueryOptions } from '@tanstack/react-query';

const publicationQueries = {
  list(query: ParsedUrlQuery): UseQueryOptions {
    return {
      queryKey: ['listPublications', query],
      queryFn: () =>
        publicationService.listPublications(
          createPublicationListRequest(query),
        ),
    };
  },
} as const;

export default publicationQueries;
```

Then we would use it, like so:

```ts
// getServerSideProps
queryClient.prefetchQuery(publicationQueries.list(query)),

// client-side
useQuery(publicationQueries.list(router.query));
useQuery({
  ...publicationQueries.list(router.query),
  keepPreviousData: true,
  staleTime: 60000,
});
```

This pattern is very similar to how we define services, but we're just defining these factory functions that generate query options instead.

Alternatively, if we really want to standardise the crap out of this, we could use the following package: https://github.com/lukemorales/query-key-factory

There's some benefit to using this package, but I'm not totally convinced we need it right away. We're probably not doing enough with React Query to warrant the additional dependency just yet. However, if you like what you see (would advise you to read the docs for some inspiration), happy to bring it in as well.

Reply:
Cool, I've done this now. Didn't use query-key-factory, but definitely something to consider later on if we use react-query in more places.